code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -35,35 +51,106 @@
+	 * This list is filled by Initialize(), if you want to change the starting wanted items, modify initial_wanteds()
+	*/
+	var/list/wanted_items
+	///Associated list of defines matched with list of phrases; phrase to be said is dealt by return_trader_phrase()
+	var/list/say_phrases = list(
+		ITEM_REJECTED_PHRASE = list(
+			"Sorry, I'm not a fan of anything you're showing me. Give me something better and we'll talk."
+		),
+		ITEM_SELLING_CANCELED_PHRASE = list(
+			"What a shame, tell me if you changed your mind."
+		),
+		ITEM_SELLING_ACCEPTED_PHRASE = list(
+		"Pleasure doing business with you."
+		),
+		INTERESTED_PHRASE = list(
+			"Hey, you've got an item that interests me, I'd like to buy it, I'll give you some cash for it, deal?"
+		),
+		BUY_PHRASE = list(
+			"Pleasure doing business with you."
+		),
+		NO_CASH_PHRASE = list(
+			"Sorry adventurer, I can't give credit! Come back when you're a little mmmmm... richer!"
+		),
+		NO_STOCK_PHRASE = list(
+			"Sorry adventurer, but that item is not in stock at the moment."
+		),
+		NOT_WILLING_TO_BUY_PHRASE = list(
+			"I don't want to buy that item for the time being, check back another time."
+		),
+		ITEM_IS_WORTHLESS_PHRASE = list(
+			"This item seems to be worthless on a closer look, I won't buy this."
+		),
+		TRADER_HAS_ENOUGH_ITEM_PHRASE = list(
+			"I already bought enough of this for the time being."
+		),
+		TRADER_LORE_PHRASE = list(
+			"Hello! I am the test trader.",
+			"Oooooooo~!"
+		)
 	)
+	///The name of the currency that is used when buying or selling items
+	var/currency_name = "credits"
+
+///Initializes the products and item demands of the trader
+/mob/living/simple_animal/hostile/retaliate/trader/Initialize()
+	. = ..()
+	restock_products()
+	renew_item_demands()
+
+///Returns a list of the starting price/quanity/fluff text about the product listings; products = initial(products) doesn't work so this exists mainly for restock_products()
+/mob/living/simple_animal/hostile/retaliate/trader/proc/initial_products()
+	return list(/obj/item/food/burger/ghost = list(200, INFINITY)
+				)
 
+///Returns a list of the starting price/quanity/fluff text about the wanted items; wanted_items = initial(wanted_items) doesn't work so this exists mainly for renew_item_demands()
+/mob/living/simple_animal/hostile/retaliate/trader/proc/initial_wanteds()
+	return list(/obj/item/ectoplasm = list(100, INFINITY, "")
+				)
+
+/**
+ * Depending on the passed parameter/override, returns a randomly picked string out of a list
+ *
+ * Do note when overriding this argument, you will need to ensure pick(the list) doesn't get supplied with a list of zero length
+ * Arguments:
+ * * say_text - (String) a define that matches the key of a entry in say_phrases
+ */
+/mob/living/simple_animal/hostile/retaliate/trader/proc/return_trader_phrase(say_text)
+	return (length(say_phrases[say_text]) ? pick(say_phrases[say_text]) : "")
```

Comment:
Why not have this be an if statement, and just ``return`` instead of ``return ""``